utilities/tokensReplacement/helper/Get-RemoteCustomParameterFileTokens.ps1

```diff
@@ -46,43 +52,42 @@ function Get-RemoteCustomParameterFileTokens {
         ## Find Token Key Vault by Name
         $TokensKeyVault = Get-AzKeyVault -VaultName $TokensKeyVaultName -ErrorAction SilentlyContinue
-        ## If Key Vault has been found, Get the Tokens
-        if ($TokensKeyVault) {
-            ## Get Tokens
-            Write-Verbose("Tokens Key Vault Found: $($TokensKeyVault.VaultName)")
-            $KeyVaultTokens = Get-AzKeyVaultSecret -VaultName $TokensKeyVault.VaultName -ErrorAction SilentlyContinue |
-                Where-Object -Property ContentType -Like '*ParameterFileToken' |
-                Where-Object -Property Name -Like "$($TokensKeyVaultSecretNamePrefix)*"
-            ## If Tokens Found
-            if ($KeyVaultTokens) {
-                Write-Verbose("Key Vault Tokens Found: $($KeyVaultTokens.count)")
-                ## Get Token Values and Add to the Returned Object
-                $KeyVaultTokens | ForEach-Object {
-                    $TokenName = $PSItem.Name
-                    $GetTokenInput = @{
-                        SecretName = $TokenName
-                        VaultName  = $TokensKeyVault.VaultName
-                    }
-                    ## Check if Token Type is 'SecureParameterFileToken'
-                    if (($PSItem.ContentType -eq 'SecureParameterFileToken')) {
-                        $TokenValue = (Get-AzKeyVaultSecret @GetTokenInput -ErrorAction SilentlyContinue).SecretValue
-                    } else {
-                        $GetTokenInput += @{ AsPlainText = $true }
-                        $TokenValue = Get-AzKeyVaultSecret @GetTokenInput -ErrorAction SilentlyContinue
-                    }
-
-                    ## Remove Prefix if Provided to Find the Token (Secret) in Key Vault
-                    if ($TokensKeyVaultSecretNamePrefix) {
-                        $TokenName = $TokenName.Replace($TokensKeyVaultSecretNamePrefix, '')
-                    }
-                    ## Add Token to Return Object
-                    $ReturnedKeyVaultTokens += [ordered]@{ Name = $TokenName; Value = $TokenValue }
-                }
+        # If no Key Vault exists. Exit
+        if (!$TokensKeyVault) {
+            Write-Verbose('No Tokens Key Vault Detected in the current Subscription Context')
+            exit
+        }
+        ## Get Tokens
+        Write-Verbose("Tokens Key Vault Found: $($TokensKeyVault.VaultName)")
+        $KeyVaultTokens = Get-AzKeyVaultSecret -VaultName $TokensKeyVault.VaultName -ErrorAction SilentlyContinue |
+            Where-Object -Property ContentType -Like '*ParameterFileToken' |
+            Where-Object -Property Name -Like "$($TokensKeyVaultSecretNamePrefix)*"
+        ## If no Tokens exist. Exit
+        if (!$TokensKeyVault) {
+            Write-Verbose("No Tokens Found using TokensKeyVaultSecretNamePrefix '$TokensKeyVaultSecretNamePrefix' In Tokens Key Vault or Service Principal does not have permissions to Token Key Vault")
+            exit
+        }
```

Comment:
Should this be if (!KeyVaultTokens) ? A few lines up you are already testing this and exiting.

Reply:
Reviewed an older commit. Re-reviewing...